Sources/SwiftDocC/Model/Rendering/RenderSectionTranslator/DeclarationsSectionTranslator.swift

```diff
@@ -114,3 +208,242 @@ struct DeclarationsSectionTranslator: RenderSectionTranslator {
         }
     }
 }
+
+fileprivate extension DeclarationRenderSection.Token {
```

Comment:
Alternatively we could change the various token properties from `let` to `var`. That way, the call site can directly modify the property instead of "cloning" the token:

```diff
- previousToken = .init(cloning: previousToken, withHighlight: nil)
+ previousToken.highlight = nil
```

```diff
- currentToken = .init(cloning: currentToken, withText: trimmedText)
+ currentToken.text = trimmedText
```

There shouldn't be any risk with using `var` for all these properties since `Token` is a struct, which means each value is its own copy and there's never any shared tokens.

Reply:
I didn't want to change the struct definition, but if we're fine with that it should clean up the code a lot more.

Reply:
I didn't make these extensions public since i was under the assumption that these tokens aren't going to be constructed much outside of Swift-DocC itself. I like having these convenience wrappers, but i'm less confident about committing to them as public API.